File: src/types/prerequisites/single/SocialStatusPrerequisite.ts
import * as DB from "tsondb/schema/dsl"
import { SocialStatusIdentifier } from "../../_Identifier.js"
import { DisplayOption } from "../DisplayOption.js"

export const SocialStatusPrerequisite = DB.TypeAlias(import.meta.url, {
  name: "SocialStatusPrerequisite",
  comment: "Requires a minimum social status.",
  type: () =>
    DB.Object({
      id: DB.Required({
        comment: "The minimum social status’s identifier.",
        type: SocialStatusIdentifier(),
      }),
      display_option: DB.Optional({
        type: DB.IncludeIdentifier(DisplayOption),
      }),
    }),
})

File: Sources/OptolithDatabaseSchema/GeneratedTypes/prerequisites/single/SocialStatusPrerequisite.swift
//
//  SocialStatusPrerequisite.swift
//  OptolithDatabaseSchema
//

import DiscriminatedEnum

/// Requires a minimum social status.
public struct SocialStatusPrerequisite: EntitySubtype {
    /// The minimum social status' identifier.
    public let id: SocialStatusIdentifier

    public let displayOption: DisplayOption?

    public init(id: SocialStatusIdentifier, displayOption: DisplayOption? = nil) {
        self.id = id
        self.displayOption = displayOption
    }

    private enum CodingKeys: String, CodingKey {
        case id = "id"
        case displayOption = "display_option"
    }
}
